### ISSUE TITLE
Link completion only works at start of line

### ISSUE THREAD

**Issue by alicekaerast** [epwalsh/obsidian.nvim#243]
### 🐛 Describe the bug

The only way I can get completion to work when creating links is when the link is the very start of a line. Otherwise it simply doesn't work. I'm a Lunarvim user and my configuration looks like:

```lua
lvim.plugins = {
  {
    "epwalsh/obsidian.nvim",
    version = "*",
    dependencies = {
      "nvim-lua/plenary.nvim",
      "hrsh7th/nvim-cmp"
    },
    opts = {
      workspaces = {
        {
          name = "notes",
          path = "/Users/Alice/notes"
        }
      },
      daily_notes = {
        folder = "Inbox/Daily",
        date_format = "%Y/%m-%B/%Y-%m-%e-%A",
        alias_format = "%Y-%m-%d",
        template = "Daily Note Template.md",
      },
      completion = {
        nvim_cmp = true,
        min_chars = 2,
      },
      templates = {
        subdir = "90-99 system/90 templates",
      },
      finder = "telescope.nvim",
    }
  }
}
```

### Versions

NVIM v0.9.4
Build type: Release
LuaJIT 2.1.1700008891

"nvim-cmp": { "branch": "main", "commit": "11102d3db12c7f8b35265ad0e17a21511e5b1e68" },
"obsidian.nvim": { "branch": "main", "commit": "5fa1ab80eeab116353f67e658c4924325b714b51" },

**Comment by linduxed** [comment]
I have the same issue.

Bisecting points to #237 being where the first bad commit (bfa94379f22fd352de7332369a5dc4f74515ecc2) is introduced.

**Comment by epwalsh** [comment]
Ugh, `nvim-cmp` is very finicky sometimes 🤦 Fixed as of 9e3709f

**Comment by linduxed** [comment]
Thank you very much for fixing the issue!
